src/picongpu/include/traits/PICToOpenPMD.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2016 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "traits/Unit.hpp"
+#include "traits/UnitDimension.hpp"
+
+#include "pmacc_types.hpp"
+#include "simulation_defines.hpp"
+
+#include <string>
+#include <vector>
+
+namespace picongpu
+{
+namespace traits
+{
+/** Reinterpret attributes for openPMD
+ *
+ * Currently, this conversion table is used to translate the PIConGPU
+ * globalCellIdx (unitless cell index) to the openPMD positionOffset (length)
+ */
+    template<typename T_Identifier>
+    struct OpenPMDName
+    {
+        std::string operator()() const
+        {
+            return T_Identifier::getName();
+        }
+    };
+
+    template<typename T_Type>
+    struct OpenPMDName<globalCellIdx<T_Type> >
+    {
+        std::string operator()() const
+        {
+            return std::string("positionOffset");
+        }
+    };
+
+    template<typename T_Identifier>
+    struct OpenPMDUnit
+    {
+        std::vector<double> operator()() const
+        {
+            return Unit<T_Identifier>::get();
+        }
+    };
+
+    template<typename T_Type>
+    struct OpenPMDUnit<globalCellIdx<T_Type> >
+    {
+        std::vector<double> operator()() const
+        {
+            std::vector<double> unit(simDim);
+            /* cell positionOffset needs two transformations to get to SI:
+               cell begin -> dimensionless scaling to grid -> SI */
+            for( uint32_t i=0; i < simDim; ++i )
+                unit[i] = cellSize[i] * UNIT_LENGTH;
+
+            return unit;
+        }
+    };
+
+    template<typename T_Identifier>
+    struct OpenPMDUnitDimension
+    {
+        std::vector<float_64> operator()() const
+        {
+            return UnitDimension<T_Identifier>::get();
+        }
+    };
+
+    template<typename T_Type>
+    struct OpenPMDUnitDimension<globalCellIdx<T_Type> >
+    {
+        std::vector<float_64> operator()() const
+        {
+            /* L, M, T, I, theta, N, J
+             *
+             * positionOffset is in meter: m
+             *   -> L
+             */
+            std::vector<float_64> unitDimension( NUnitDimension, 0.0 );
+            unitDimension.at(SIBaseUnits::length) = 1.0;
```

Comment:
yes, `globalCellIdx` is an actual index and I will change `speciesAttributes.unitless` to make it unitless.